k8s/k8s.go

```diff
@@ -960,6 +976,22 @@ func (k *k8sOps) GetPodsByOwner(ownerName string, namespace string) ([]v1.Pod, e
 	return result, nil
 }
 
+func (k *k8sOps) SearchPodsByUID(uid string, namespace string) (*v1.Pod, error) {
+	pods, err := k.GetPods(namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	pUID := types.UID(uid)
+	for _, pod := range pods.Items {
+		if pod.UID == pUID {
+			return &pod, nil
+		}
+	}
+
+	return nil, nil
```

Comment:
Should we return an error if the pod is not found?

Reply:
We could -- but in that case we better be consistent, and make all other Get/Search -functions (ie. `GetPodsByOwner`) also return the `ErrPodsNotFound` error when stuff isn't found;  note that this also means people will need to check against public Error-object (ie. `if err == ErrPodsNotFound`) instead of empty return values.